### PR TITLE
fix(sst): capacity estimate

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

We set capacity no more than the total size of the input files to reduce memory usage. But the meta info maybe reserve some other space so that it would generate one small file.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
- https://github.com/risingwavelabs/risingwave/issues/5350

### CODE REVIEW

File: src/storage/src/hummock/compactor/mod.rs [hunk 1]
```diff
@@ -801,7 +801,7 @@ async fn generate_splits(compact_task: &mut CompactTask, context: Arc<Context>)
         .sum::<u64>();
 
     let sstable_size = (context.options.sstable_size_mb as u64) << 20;
-    if compaction_size > sstable_size {
+    if compaction_size > sstable_size * 2 {
```
Comment (Li0k) [inline src/storage/src/hummock/compactor/mod.rs:804]: Can we use `sstable_size * max_sub_compaction` to ensure the min_size of splits larger than single `sstable_size`

**Review by Li0k — APPROVED** [review]
Rest LGTM

**Comment by Little-Wallace** [comment]
This branch:
image

main branch:
image

**Comment by Little-Wallace** [comment]
There is still another problem left that is if `CapacitySplitTableBuilder` may left a small data in an independent file, which also cause a small file.

**Pull request merged**